File: examples/url.py
import Pyjo.URL
from Pyjo.String.Unicode import u

# 'ssh+git://git@github.com/dex4er/Pyjo.git'
url = Pyjo.URL.new('https://github.com/dex4er/Pyjo')
print(url.set(scheme='ssh+git', userinfo='git', path=u(url.path) + '.git'))

# 'http://metacpan.org/search?q=Mojo::URL&size=20'
print(Pyjo.URL.new('http://metacpan.org/search')
      .set(query={'q': 'Mojo::URL', 'size': 20}))

File: examples/url.pl
use v5.14;

use Mojo::URL;

my $url = Mojo::URL->new('https://github.com/dex4er/Pyjo');
say $url->scheme('ssh+git')->userinfo('git')->path($url->path.'.git');

say Mojo::URL->new('https://metacpan.org/search')->query(q => 'Mojo::URL', size => 20);
